src/App.tsx

```diff
@@ -1,25 +1,141 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import { useEffect, useState } from "react";
+
+import {
+  Container,
+  CssBaseline,
+  Grid,
+  LinearProgress,
+} from "@material-ui/core";
+import Header from "./component/Header";
+
+//styles
+import "./App.css";
+
+import CoverPost from "./component/CoverPost";
+import MediaCard from "./component/CardComponent";
+import Footer from "./component/Footer";
+
+export type CardItemType = {
+  id: number;
+  category: string;
+  description: string;
+  image: string;
+  price: number;
+  title: string;
+};
+
+export type ShoppingCart = {
+  id: number;
+  itemQty: number;
+  itemName: string;
+  image: string;
+  price: number;
+};
 
 function App() {
+  const [sideDrawer, setsideDrawer] = useState(false);
+  const [cart, setcart] = useState([] as ShoppingCart[]);
+  const [appState, setAppState] = useState({
+    loading: false,
+    data: [],
+  } as {
+    loading: boolean;
+    data: CardItemType[] | [];
+  });
+
+  useEffect(() => {
+    setAppState({ loading: true, data: [] });
+    const apiUrl = "https://fakestoreapi.com/products";
+    fetch(apiUrl)
+      .then((res) => res.json())
+      .then(async (shopingData) =>
+        setAppState({ loading: false, data: shopingData })
+      )
+      .catch((err) => {
+        console.log(err);
+        setAppState({ loading: false, data: [] });
```

Comment:
We should show some error to user so they know that there was some error in fetching and not that there are no data to be displayed

Reply:
Implemented